larpandoracontent/LArHelpers/LArInteractionTypeHelper.cc

```diff
@@ -44,17 +44,130 @@ LArInteractionTypeHelper::InteractionType LArInteractionTypeHelper::GetInteracti
         else return COSMIC_RAY_OTHER;
     }
 
-    if ((1 == mcPrimaryList.size()) && LArMCParticleHelper::IsBeamParticle(mcPrimaryList.front()))
+    if (!testBeamHierarchyMode)
```

Comment:
Suggest separate these two helpers into GetInteractionType and GetHierarchyInteractionType (or a better name), then use the algorithm logic to decide which to call. Part of a general pattern of teasing apart implementation.